File: TrabajoAcademicoNet/SoftProgModelo/VentaModelo/OrdenCompra.cs
using PUCP.Inf30.SoftProg.Modelo.BaseModelo;
using PUCP.Inf30.SoftProg.Modelo.Venta.VentaUtils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace PUCP.Inf30.SoftProg.Modelo.VentaModelo
{
    public class OrdenCompra:Registro
    {
        public List<LineaOrdenCompra> LineasOrden {  get; set; }
        public DateTime FechaCreacion { get; set; }
        public double TotalParcial { get; set; }
        public double TotalFinal { get; set; }
        public double DescuentoTotal { get; set; }
        public int IdDetalleEnvio { get; set; }
        public EstadoOrdenCompra Estado;

        public OrdenCompra()
        {
            LineasOrden = new List<LineaOrdenCompra>();
        }

        public OrdenCompra(List<LineaOrdenCompra> lineasOrden, DateTime fechaCreacion, double totalParcial, 
            double totalFinal, double descuentoTotal, int idDetalleEnvio, EstadoOrdenCompra estado)
        {
            this.LineasOrden = LineasOrden;
            this.FechaCreacion = fechaCreacion;
            this.TotalParcial = totalParcial;
            this.TotalFinal = totalFinal;
            this.DescuentoTotal = descuentoTotal;
            this.IdDetalleEnvio = idDetalleEnvio;
            this.Estado = estado;
        }

        public OrdenCompra(List<LineaOrdenCompra> lineasOrden, DateTime fechaCreacion, double totalParcial, 
            double totalFinal, double descuentoTotal, int idDetalleEnvio, EstadoOrdenCompra estado, int id, bool activo) :base(id,activo)
        {
            this.LineasOrden = lineasOrden;
            this.FechaCreacion = fechaCreacion;
            this.TotalParcial = totalParcial;
            this.TotalFinal = totalFinal;
            this.DescuentoTotal = descuentoTotal;
            this.IdDetalleEnvio = idDetalleEnvio;
            this.Estado = estado;
        }
        public String getEstadoString()
        {
            String estado=" ";
            switch (this.Estado)
            {
                case EstadoOrdenCompra.PAGADO:
                    estado = "PAGADO";
                    break;
                case EstadoOrdenCompra.EN_PREPARACION:
                    estado = "EN_PREPARACION";
                    break;
                case EstadoOrdenCompra.ENVIADO:
                    estado = "ENVIADO";
                    break;
                case EstadoOrdenCompra.ENTREGADO:
                    estado = "ENTREGADO";
                    break;
                case EstadoOrdenCompra.CANCELADO:
                    estado = "CANCELADO";
                    break;
                case EstadoOrdenCompra.REEMBOLSADO:
                    estado = "REEMBOLSADO";
                    break;
                case EstadoOrdenCompra.EXPIRADO:
                    estado = "EXPIRADO";
                    break;
            }
            return estado;
        }

        public void setEstadoString(String estado)
        {
            switch (estado)
            {
                case "PAGADO":
                    this.Estado = EstadoOrdenCompra.PAGADO;
                    break;
                case "EN_PREPARACION":
                    this.Estado = EstadoOrdenCompra.EN_PREPARACION;
                    break;
                case "ENVIADO":
                    this.Estado = EstadoOrdenCompra.ENVIADO;
                    break;
                case "ENTREGADO":
                    this.Estado = EstadoOrdenCompra.ENTREGADO;
                    break;
                case "CANCELADO":
                    this.Estado = EstadoOrdenCompra.CANCELADO;
                    break;
                case "REEMBOLSADO":
                    this.Estado = EstadoOrdenCompra.REEMBOLSADO;
                    break;
                case "EXPIRADO":
                    this.Estado = EstadoOrdenCompra.EXPIRADO;
                    break;
            }
        }
    }
}

File: SoftProgModelo/src/main/java/pe/edu/pucp/inf30/softprog/modelo/venta/OrdenCompra.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.inf30.softprog.modelo.venta;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import pe.edu.pucp.inf30.softprog.modelo.base.Registro;
import pe.edu.pucp.inf30.softprog.modelo.persona.Cliente;
import pe.edu.pucp.inf30.softprog.modelo.venta.utils.EstadoOrdenCompra;

/**
 *
 * @author Cristhian Horacio
 */
public class OrdenCompra extends Registro {

    private List<LineaOrdenCompra> lineasOrden;
    private Date fechaCreacion;
    private double totalParcial;
    private double totalFinal;
    private double descuentoTotal;
    private DetalleEnvio detalleEnvio;
    private Cliente cliente;
    private EstadoOrdenCompra estado;

    public OrdenCompra() {
        lineasOrden = new ArrayList<>();
    }

    public OrdenCompra(List<LineaOrdenCompra> lineasOrden, Date fechaCreacion, double totalParcial, double totalFinal, double descuentoTotal, DetalleEnvio detalleEnvio, Cliente cliente, EstadoOrdenCompra estado) {
        this.lineasOrden = lineasOrden;
        this.fechaCreacion = fechaCreacion;
        this.totalParcial = totalParcial;
        this.totalFinal = totalFinal;
        this.descuentoTotal = descuentoTotal;
        this.detalleEnvio = detalleEnvio;
        this.cliente = cliente;
        this.estado = estado;
    }

    public OrdenCompra(List<LineaOrdenCompra> lineasOrden, Date fechaCreacion, double totalParcial, double totalFinal, double descuentoTotal, DetalleEnvio detalleEnvio, Cliente cliente, EstadoOrdenCompra estado, int id, boolean activo) {
        super(id, activo);
        this.lineasOrden = lineasOrden;
        this.fechaCreacion = fechaCreacion;
        this.totalParcial = totalParcial;
        this.totalFinal = totalFinal;
        this.descuentoTotal = descuentoTotal;
        this.detalleEnvio = detalleEnvio;
        this.cliente = cliente;
        this.estado = estado;
    }

    

    public List<LineaOrdenCompra> getLineasOrden() {
        return lineasOrden;
    }

    public void setLineasOrden(List<LineaOrdenCompra> lineasOrden) {
        this.lineasOrden = lineasOrden;
    }

    public Date getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(Date fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }

    public double getTotalParcial() {
        return totalParcial;
    }

    public void setTotalParcial(double totalParcial) {
        this.totalParcial = totalParcial;
    }

    public double getTotalFinal() {
        return totalFinal;
    }

    public void setTotalFinal(double totalFinal) {
        this.totalFinal = totalFinal;
    }

    public double getDescuentoTotal() {
        return descuentoTotal;
    }

    public void setDescuentoTotal(double descuentoTotal) {
        this.descuentoTotal = descuentoTotal;
    }

    public EstadoOrdenCompra getEstado() {
        return estado;
    }

    public void setEstado(EstadoOrdenCompra estado) {
        this.estado = estado;
    }

    public String getEstadoString() {
        return switch (this.estado) {
            case PAGADO ->
                "PAGADO";
            case EN_PREPARACION ->
                "EN_PREPARACION";
            case ENVIADO ->
                "ENVIADO";
            case ENTREGADO ->
                "ENTREGADO";
            case CANCELADO ->
                "CANCELADO";
            case REEMBOLSADO ->
                "REEMBOLSADO";
            case EXPIRADO ->
                "EXPIRADO";
        };
    }

    public void setEstadoString(String estado) {
        switch (estado) {
            case "PAGADO" ->
                this.estado = EstadoOrdenCompra.PAGADO;
            case "EN_PREPARACION" ->
                this.estado = EstadoOrdenCompra.EN_PREPARACION;
            case "ENVIADO" ->
                this.estado = EstadoOrdenCompra.ENVIADO;
            case "ENTREGADO" ->
                this.estado = EstadoOrdenCompra.ENTREGADO;
            case "CANCELADO" ->
                this.estado = EstadoOrdenCompra.CANCELADO;
            case "REEMBOLSADO" ->
                this.estado = EstadoOrdenCompra.REEMBOLSADO;
            case "EXPIRADO" ->
                this.estado = EstadoOrdenCompra.EXPIRADO;
        }
    }

    public DetalleEnvio getDetalleEnvio() {
        return detalleEnvio;
    }

    public void setDetalleEnvio(DetalleEnvio detalleEnvio) {
        this.detalleEnvio = detalleEnvio;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    

}
